spec/services/resync_service_spec.rb

```diff
@@ -50,5 +50,37 @@
         ResyncService.call(document)
       end
     end
+
+    context "when the live edition is withdrawn" do
+      let(:edition) { create(:edition, :withdrawn) }
+      let(:explanation) { "explanation" }
+
+      before do
+        stub_any_publishing_api_unpublish
+        allow(PreviewAssetService).to receive(:call)
+        allow(PublishAssetService).to receive(:call)
+      end
+
+      it "withdraws the edition" do
+        allow(GovspeakDocument)
+          .to receive(:new)
+          .and_return(instance_double(GovspeakDocument, payload_html: explanation))
+
+        withdraw_params = {
+          type: "withdrawal",
+          explanation: explanation,
+          locale: edition.locale,
+        }
+
+        expect(GovspeakDocument)
+          .to receive(:new)
+          .with(edition.status.details, edition)
```

Comment:
You can probably change this to:

```
        expect(GovspeakDocument)
          .to receive(:new)
          .with(edition.status.details, edition)
          .and_return(instance_double(GovspeakDocument, payload_html: explanation))
```
and remove the need for the allow on line 65